neuvue_project/workspace/plugins.py

```diff
@@ -87,11 +91,75 @@ def modify_state(self, state: Dict[str, Any]) -> PluginOutput:
             additional_info={"test": "This is a test plugin."}
         )
 
-#                           Add new plugins here
+class NeurdC2SkeletonPointsPlugin(NeuroglancerPlugin):
+    """
+    This plugin queries NEURD for the skeletons of all C2 seg IDs listed and
+    adds them to the ng state as a new annotation layer.
+    """
+
+    def modify_state(self, state: Dict[str, Any]) -> PluginOutput:
+
+        # Get seg ids to query
+        C2_SEG_LAYER = "precomputed://gs://h01-release/data/20210601/c2"
+        seg_ids = []
+        for layer in state["layers"]:
+            if layer["source"] == C2_SEG_LAYER:
+                seg_ids = layer["segments"]
+        position = state["position"]
+
+        # Query for skeleton
+        BASE_URL = settings.NEURD_LAMBDA_URL
+        skel_list = []
+        for seg_id in seg_ids:
+            if '!' not in seg_id:
+                request_url = f"{BASE_URL}/skeleton/{seg_id}"
+                try:
+                    response = requests.get(request_url)
+                    # Raise exception if request is unsuccessful
+                    response.raise_for_status()
+                    # Process the successful response
+                    skel_list.extend(json.loads(response.text)["skeleton_points"])
+                except Exception as e:
+                    print(f"An unexpected error occurred: {e}")
+                    return PluginOutput(
+                        modified_state=state,
+                        status_code=500,
+                        message=f"An unexpected error occurred: {e}",
+                        additional_info={}
+                    )
+        
+        # Process the skeleton points, converting them from nm to voxels
+        RESOLUTION = [8,8,33]
```

Comment:
might be nitpicky buy maybe these semi-global variables like resolution and neurd_url could possibly be defined as class variables outside of modify_state to make it easy configurable if for future versions we break out into smaller functions or make updates to this plugin (neurd_h01_plugin, neurd_minnie35_plugin, etc)